allennlp/modules/span_extractors/locally_normalised_span_extractor.py

```diff
@@ -0,0 +1,96 @@
+import torch
+from overrides import overrides
+
+from allennlp.common.params import Params
+from allennlp.modules.span_extractors.span_extractor import SpanExtractor
+from allennlp.modules.time_distributed import TimeDistributed
+from allennlp.nn import util
+
+@SpanExtractor.register("locally_normalised")
+class LocallyNormalisedSpanExtractor(SpanExtractor):
+    """
+    Computes span representations by generating an unnormalized attention score for each
+    word in the document, compute distributions over every span with respect to these
+    scores by normalising the attention scores for words inside the span.
+
+    Given these attention distributions over every span, weight the corresponding vector
```

Comment:
I'd recommend declarative instead of imperative descriptions here.  "Given ..., this module weights the", or whatever your favorite way of doing that is.